src/types/keyboard.rs

```diff
@@ -0,0 +1,114 @@
+use std::fmt;
+use wlroots_sys::{wlr_keyboard, wlr_keyboard_get_modifiers, wlr_keyboard_led,
+                  wlr_keyboard_led_update, wlr_keyboard_modifier, wlr_keyboard_set_keymap,
+                  xkb_keymap};
+
+#[derive(Debug)]
+pub struct Keyboard {
+    keyboard: *mut wlr_keyboard,
+}
+
+impl Keyboard {
+    pub(crate) fn new(kb_pointer: *mut wlr_keyboard) -> Self {
+        Keyboard { keyboard: kb_pointer }
+    }
+
+    pub(crate) unsafe fn to_ptr(&self) -> *mut wlr_keyboard {
+        self.keyboard
+    }
+
+    // TODO: Implement keymap wrapper?
+    pub fn set_keymap(&mut self, keymap: *mut xkb_keymap) {
+        unsafe {
+            wlr_keyboard_set_keymap(self.keyboard, keymap);
+        }
+    }
+
+    pub fn led_update(&mut self, leds: KeyboardLed) {
+        unsafe {
+            wlr_keyboard_led_update(self.keyboard, leds.state());
+        }
+    }
+
+    pub fn get_modifiers(&self) -> KeyboardModifier {
+        unsafe { KeyboardModifier::new(wlr_keyboard_get_modifiers(self.keyboard)) }
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct KeyboardLed {
```

Comment:
Replace this struct by using the `bitflags` crate to generate this type. It's more clear what it is that way